include/lsst/afw/geom/SkyWcs.h

```diff
@@ -94,158 +100,391 @@ Constructors must set the following properties of the contained ast::FrameSet:
+     *
+     * @warning AST cannot yet write out SIP terms, so if precise=true this will fail for a TAN-SIP WCS
+     * with significant distortion.
+     *
+     * @throws lsst::pex::exceptions::RuntimeError if precise is true and AST cannot represent
+     * this WCS as a FITS WCS to sufficient precision.
+     */
+    std::shared_ptr<daf::base::PropertyList> getFitsMetadata(bool precise) const;
 
-    This is the most general constructor; it can be used to define any celestial WCS
-    as long as it contains the @ref skywcs_frames "required frames".
+    /**
+     * Get the pixel scale at the specified pixel position
+     *
+     * The scale is the square root of the area of the specified pixel on the sky.
+     *
+     * @warning Unlike getPixelScale() the value is not cached, even if pixel = pixel origin.
+     */
+    Angle getPixelScale(Point2D const &pixel) const;
 
-    @throw lsst::pex::exceptions::InvalidParameterError if `frameSet` is missing
-    any of the @ref skywcs_frames "required frames".
-    */
-    explicit SkyWcs(ast::FrameSet const &frameSet);
+    /**
+     * Get the pixel scale at the pixel origin
+     *
+     * The scale is the square root of the area of the specified pixel on the sky.
+     *
+     * The value is cached, so this is a cheap call.
+     */
+    Angle getPixelScale() const { return _pixelScaleAtOrigin; };
 
-    ~SkyWcs(){};
+    /**
+     * Get CRPIX, the pixel origin, using the LSST convention
+     */
+    Point2D getPixelOrigin() const { return _pixelOrigin; };
 
     /**
-    Get the pixel scale at the specified pixel position
+     * Get CRVAL, the sky origin or celestial fiducial point
+     */
+    coord::IcrsCoord getSkyOrigin() const;
 
-    The scale is the square root of the area of the specified pixel on the sky.
-    */
-    Angle getPixelScale(Point2D const &pixel) const;
+    /**
+     * Get the 2x2 CD matrix at the specified pixel position
+     *
+     * The elements are in degrees
+     */
+    Eigen::Matrix2d getCdMatrix(Point2D const &pixel) const;
 
     /**
-    Get the pixel scale at the pixel origin
+     * Get the 2x2 CD matrix at the pixel origin
+     *
+     * The elements are in degrees
+     */
+    Eigen::Matrix2d getCdMatrix() const;
 
-    The scale is the square root of the area of the specified pixel on the sky.
-    */
-    Angle getPixelScale() const;
+    /**
+     * Get a local TAN WCS approximation to this WCS at the specified pixel position
+     */
+    std::shared_ptr<SkyWcs> getTanWcs(Point2D const &pixel) const;
 
     /**
-    Get CRPIX, the pixel origin, using the LSST convention
-    */
-    Point2D getPixelOrigin() const;
+     * Get the contained FrameDict
+     *
+     * The forward transform goes from pixels (using the LSST zero convention)
+     * to sky ICRS RA, Dec (in radians)
+     */
+    std::shared_ptr<const ast::FrameDict> getFrameDict() const;
```

Comment:
I'm not sure how much this helps, but I would like to make it easy for jointcal to construct one SkyWcs from pieces of another, as it can be rather a lot of work to construct one from scratch.